### ISSUE TITLE
Do we have to make y_train 3-d tensor/array if we apply return_sequences=true?

### ISSUE THREAD

**Issue by Kyubyong** [fchollet/keras#3336]
For example, assume we want to make a character-rnn model with Keras. We have 100 sentences in total, and each sentence consists of 10 characters, and there are 30 characters in total. Since we want to predict the next character for every time step, we need to set return_suquences=True, right? Then I guess, we should make y_train 3-d array of the shape (100, 10, 30). The last axis is one-hot encoded. I'm wondering if we really should do so because one-hot encoding is somewhat cumbersome and computationally expensive.

**Comment by ymcui** [comment]
in `categorical_crossentropy`, you should arrange your output as 3D-tensor.
but here is another choice: `sparse_categorical_crossentropy`, which enables you to directly use `word index`.
check this: http://keras.io/objectives/
